app/src/main/java/com/google/samples/apps/sunflower/PlantDetailFragment.kt

```diff
@@ -56,11 +57,41 @@ class PlantDetailFragment : Fragment() {
         val binding = DataBindingUtil.inflate<FragmentPlantDetailBinding>(
                 inflater, R.layout.fragment_plant_detail, container, false).apply {
             viewModel = plantDetailViewModel
-            lifecycleOwner = this@PlantDetailFragment
+            lifecycleOwner = viewLifecycleOwner
             fab.setOnClickListener { view ->
+                hideAppBarFab(fab)
                 plantDetailViewModel.addPlantToGarden()
                 Snackbar.make(view, R.string.added_plant_to_garden, Snackbar.LENGTH_LONG).show()
             }
+
+            // scroll change listener begins at Y = 0 when image is fully collapsed
+            plantDetailScrollview.setOnScrollChangeListener(NestedScrollView.OnScrollChangeListener { _, _, scrollY, _, _ ->
+                // user scrolled past image to height of toolbar
+                val showToolbar = scrollY > toolbar.height
+
+                // set title programatically so not shown by default
+                toolbarLayout.title = viewModel?.plant?.value?.name
+
+                // use shadow animator to add elevation when toolbar should appear
+                appbar.isActivated = showToolbar
+
+                // plant name in body is off screen, so show plant name in toolbar
+                toolbarLayout.isTitleEnabled = showToolbar
+            })
+
+            toolbar.setNavigationOnClickListener { view ->
+                view.findNavController().navigateUp()
+            }
+
+            toolbar.setOnMenuItemClickListener { item ->
+                when (item.itemId) {
+                    R.id.action_share -> {
+                        createShareIntent()
+                        true
+                    }
+                    else -> false
+                }
+            }
         }
 
         plantDetailViewModel.plant.observe(this) { plant ->
```

Comment:
I know this line isn't related with the PR but caused me a curiosity about _observe_ nullable livedata. After read a little about that I found the following article, https://proandroiddev.com/nonnull-livedata-with-kotlin-extension-26963ffd0333 where expose 2 ways to fix that.

In our case is a little different because we want to keep nullable but avoiding the ide warning, relating no optional checks, for that maybe it's a good approach to create an extension like this, or with different name

```java
fun <T> LiveData<T>.observe(owner: LifecycleOwner, observer: (t: T?) -> Unit) {
        this.observe(owner, observer)
    }   
```

Reply:
Interesting - let's continue the discussion in a new Issue?